packages/thumbprint-react/components/Tooltip/test.jsx

```diff
@@ -552,82 +252,458 @@ describe('touch devices', () => {
+                        <button
+                            ref={ref}
+                            onMouseEnter={onMouseEnter}
+                            onClick={onClick}
+                            onFocus={onFocus}
+                            onMouseLeave={onMouseLeave}
+                            onBlur={onBlur}
+                            aria-label={ariaLabel}
+                            type="button"
+                        >
+                            Duck
+                        </button>
+                    )}
+                </Tooltip>
+            </div>,
         );
+        expect(jestOnClick).toHaveBeenCalledTimes(0);
+        wrapper.find('button').simulate('mouseenter');
+        expect(jestOnClick).toHaveBeenCalledTimes(0);
+
+        act(() => {
+            jest.runAllTimers();
+        });
+
+        wrapper
+            // Need to manually call update here because setState is called asynchronously in the
+            // show method.
+            .update()
+            .find('[role="tooltip"]')
+            .simulate('click');
+        expect(jestOnClick).toHaveBeenCalledTimes(1);
+    });
 
-        const button = wrapper.find('button');
-
-        wrapper.setState({ isOpen: true });
-        expect(wrapper.state('isOpen')).toBe(true);
+    describe('non-touch devices', () => {
+        test('mouseenter and mouseleave open and close the tooltip', () => {
+            const wrapper = mount(
+                <Tooltip text="Goose" container="inline">
+                    {({ ref, onMouseEnter, onClick, onFocus, onMouseLeave, onBlur, ariaLabel }) => (
+                        <button
+                            ref={ref}
+                            onMouseEnter={onMouseEnter}
+                            onClick={onClick}
+                            onFocus={onFocus}
+                            onMouseLeave={onMouseLeave}
+                            onBlur={onBlur}
+                            aria-label={ariaLabel}
+                            type="button"
+                        >
+                            Duck
+                        </button>
+                    )}
+                </Tooltip>,
+            );
+
+            const button = wrapper.find('button');
+
+            expect(wrapper.find('[role="tooltip"]').exists()).toBe(false);
+
+            button.simulate('mouseenter');
+            expect(wrapper.find('[role="tooltip"]').exists()).toBe(false);
+
+            act(() => {
+                jest.runAllTimers();
+            });
+            wrapper.update();
+            expect(wrapper.find('[role="tooltip"]').exists()).toBe(true);
+
+            button.simulate('mouseleave');
+            act(() => {
+                jest.runAllTimers();
+            });
+            wrapper.update();
+            expect(wrapper.find('[role="tooltip"]').exists()).toBe(false);
+        });
+
+        test('focus and blur open and close the tooltip', () => {
```

Comment:
Why remove the backticks? Makes it easier to know that `focus` and `blur` are events but "open" and "close" are verbs.

Reply:
Couple of reasons:

- Makes it harder to Cmd+f for the test name to jump to the failing test after seeing it in the terminal
- Breaks terminal commands that include the test name. For example running

  ```
  yarn test:jest test.jsx --testNamePattern "`mouseleave` closes the tooltip 
  immediately if closeDelayLength is 0"
  ```

  gives the error `zsh: command not found: mouseleave`.